src/include/simeng/Instruction.hh

```diff
@@ -33,14 +33,14 @@ class Instruction {
 
   /** Override the specified source register with a renamed physical register.
    */
-  virtual void renameSource(uint8_t i, Register renamed) = 0;
+  virtual void renameSource(uint16_t i, Register renamed) = 0;
 
   /** Override the specified destination register with a renamed physical
    * register. */
-  virtual void renameDestination(uint8_t i, Register renamed) = 0;
+  virtual void renameDestination(uint16_t i, Register renamed) = 0;
```

Comment:
Unsure of why this change to the type has been made throughout. How many registers could be contained within e.g. the destinationRegisters vector

Reply:
@FinnWilkinson Was this for the SME instructions with matrix destination vectors and implicit registers?

Reply:
Yes this is for SME. As each row of the ZA matrix is treated like a vector register in SimEng, supposing a 2048-bit vector length you could have up to 256 matrix rows. Add on top of this other operands, you could need up to approx 264 "registers" (i.e. SimEng's internal representation, **NOT** architectural registers)

Reply:
Thanks for the clarification